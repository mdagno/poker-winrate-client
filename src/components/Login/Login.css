#login-form {
  display: flex;
  flex-direction: column;
  color: #222;
  width: 30%;
  margin-left: auto;
  margin-right: auto;
  background-color: #ccc;
  padding: 1em;
  border: 1px outset white;
  border-radius: 5px;
  font-size: 0.8em;
  margin-top: 5%;
}

.login-details {
  margin: 1em;
  padding: 0.5em;
}

#login-button {
  margin-left: auto;
  margin-right: auto;
  width: 30%;
  margin-top: 3em;
  margin-bottom: 3em;
}

#login-header {
  margin: 0;
  margin-left: auto;
  margin-right: auto;
  font-size: 2em;
  padding: 0;
}

#login-form a {
  text-align: center;
}