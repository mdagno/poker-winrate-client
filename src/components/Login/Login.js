import React from 'react';
import ApiContext from '../../ApiContext';
import AuthService from '../../services/auth-service';
import ApiService from '../../services/api-service';
import './Login.css';
import { Link } from 'react-router-dom';

export default class Login extends React.Component {
  static contextType = ApiContext;
  state = {
    username: '',
    password: '',
  }


  handleLoginSuccess = () => {
    const { location, history } = this.props
    console.log(location.state)
    const destination = (location.state || {}).from || '/'
    history.push(destination)
  }

  submitLogin = e => {
    e.preventDefault();
    const { username, password } = e.target;

    AuthService.postLogin({
      username: username.value,
      password: password.value
    })
      .then(res => {
        username.value = '';
        password.value = '';
        this.context.processLogin(res.authToken)
        this.props.history.push('/')
      })
      .catch(res => {
        this.setState({
          error: res.error
        })
      })
  }

  handleInputChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  render() {
    return (
      <form id="login-form" onSubmit={e => this.submitLogin(e)}>
        <h3 id="login-header">Login</h3>
        <label htmlFor="username" className="login-details">Username</label>
        <input type="text" className="login-details" name="username" onChange={e => this.handleInputChange(e)}></input>
        <label htmlFor="password"className="login-details">Password</label>
        <input type="password" className="login-details" name="password" onChange={e => this.handleInputChange(e)}></input>
        <button type="submit" id="login-button">Login</button>
        <Link to="/register">Don't have an account?</Link>
      </form>
    )
  }
}