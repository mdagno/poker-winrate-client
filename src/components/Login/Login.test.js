import React from 'react';
import Login from './Login';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';

describe('Login component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(     
    <BrowserRouter>
      <Login />
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Login renders correctly', () => {
    const tree = renderer
      .create(    
      <BrowserRouter>
        <Login />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})


