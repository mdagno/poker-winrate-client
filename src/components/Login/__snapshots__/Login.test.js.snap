// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component Login renders correctly 1`] = `
<form
  id="login-form"
  onSubmit={[Function]}
>
  <h3
    id="login-header"
  >
    Login
  </h3>
  <label
    className="login-details"
    htmlFor="username"
  >
    Username
  </label>
  <input
    className="login-details"
    name="username"
    onChange={[Function]}
    type="text"
  />
  <label
    className="login-details"
    htmlFor="password"
  >
    Password
  </label>
  <input
    className="login-details"
    name="password"
    onChange={[Function]}
    type="password"
  />
  <button
    id="login-button"
    type="submit"
  >
    Login
  </button>
  <a
    href="/register"
    onClick={[Function]}
  >
    Don't have an account?
  </a>
</form>
`;
